src/main/java/uk/gov/justice/digital/hmpps/whereabouts/config/JmsConfig.kt

```diff
@@ -0,0 +1,64 @@
+package uk.gov.justice.digital.hmpps.whereabouts.config
+
+import com.amazon.sqs.javamessaging.ProviderConfiguration
+import com.amazon.sqs.javamessaging.SQSConnectionFactory
+import com.amazonaws.auth.AWSStaticCredentialsProvider
+import com.amazonaws.auth.BasicAWSCredentials
+import com.amazonaws.client.builder.AwsClientBuilder.EndpointConfiguration
+import com.amazonaws.services.sqs.AmazonSQS
+import com.amazonaws.services.sqs.AmazonSQSAsync
+import com.amazonaws.services.sqs.AmazonSQSAsyncClientBuilder
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Primary
+import org.springframework.jms.annotation.EnableJms
+import org.springframework.jms.config.DefaultJmsListenerContainerFactory
+import org.springframework.jms.support.destination.DynamicDestinationResolver
+import javax.jms.Session
+
+@Configuration
+@EnableJms
+@ConditionalOnProperty(name = ["sqs.provider"])
+open class JmsConfig {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @Bean
+  open fun jmsListenerContainerFactory(awsSqs: AmazonSQS?): DefaultJmsListenerContainerFactory {
+    val factory = DefaultJmsListenerContainerFactory()
+    factory.setConnectionFactory(SQSConnectionFactory(ProviderConfiguration(), awsSqs))
+    factory.setDestinationResolver(DynamicDestinationResolver())
+    factory.setConcurrency("3-10")
```

Comment:
might be better to set concurrency to 1 - that will give 4 listeners in live anyway